dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -33,10 +38,18 @@
 {% macro spark_build_snapshot_staging_table(strategy, sql, target_relation) %}
     {% set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' %}
 
-    {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
-                                                  schema=target_relation.schema,
-                                                  database=none,
-                                                  type='view') -%}
+    {% if target_relation.is_iceberg %}
+      {# iceberg catalog does not support create view, but regular spark does. We removed the catalog and schema #}
+      {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                    schema=none,
```

Comment:
If I understand the comments correct, `schema` isn't a thing for iceberg. If that's the case, can we just set `target_relation.schema=None` where it's defined, instead of overwriting it here?

Reply:
I am so hesitant to do this perhaps because I lack knowledge on exactly what that does. I don't see why mutating state like that would benefit us.

Reply:
Okay, I looked into this in depth. `target.schema` is set to `default` or in other words, the database that I created in the spark cluster by `CREATE DATABASE default` as part of my test env setup. 

What we call a schema, Iceberg calls a database. The `target.database` param is already `none`, which further corroborates what I stated before. Temporary tables (really, they're views) don't actually persist in memory. Hence, they have no schema or database.

Hence, I concur with Fokko's earlier comment and the macro code here. We shouldn't mutate the state here, not at least until we have a much clearer picture as to what is occurring with these objects in spark across different compute engines (as of now unclear what trends are present if any).

tl;dr I really don't think we can / should change this.